Update Read_and_Write_Excel.java

antinos opened the pull request:
When I tried your version (actually the dropbox file you sent me a while back), ImageJ spits out an IllegalArgumentException:
```java
java.lang.IllegalArgumentException: Column not defined: 3
```
This problem was familiar to me from my own meddling. As you can see, I changed numColumn handling to mitigate the suspected ImageJ bug. It may just be my usage though, so feel free to school me.

antinos committed:
Update Read_and_Write_Excel.java

numColumn change to mitigate suspected ImageJ bug.

antinos commented:
Firstly, can I just say... now I've had a chance to try your version, I love the 'file_mode' argument. The big data issue is only something I really stumbled across when I revisited the plugin in the last week. The datastream overhead issues sent me in 'search engine loops' trying to overcome the (probably) xmlbeans issues and the inherent restraints of Apache poi workbook instantiation (File vs inputstream).
Secondly, please excuse me if I don't follow GitHub etiquette. I'm teaching myself as I go along.

bkromhout commented:
I'll be sure to take a look at this after work this evening, but I would recommend trying the distribution you can get through ImageJ's updater rather than the Dropbox files (they're probably out of date)... You will probably have to uncheck your own distribution in there for that to work by the way, I found out they won't coexist side by side, it makes the updater confused.

If it still isn't working then, while I'm happy to merge this PR, I'm unfortunately not going to be able to continue with upkeep on this plugin in the long run. It started as yours, and I will will not be at all offended if you just make it your own again :)

Finally, I'm glad you like the file mode additions... They are a bit hacky to be honest (just in the way that I'm storing the file reference internally... That ImageJ API is definitely not *meant* to be used to store file references), but they have definitely been battle tested... The person I implemented them for used them for macros with a ton of data.

antinos commented:
Thanks for the info and for the permission to continue managing the release(s) with your re-coding. If it's okay, I intend to add you as an author in v1.1.0. I've already put together a new Youtube tutorial, which is currently unlisted:
https://youtu.be/dLkoB25MTIY
in which I state your efforts in adding the features and cleaning up the code. If you would prefer I don't make this video public or have any other recommendations for how to indicate your involvement, then just let me know. I've done very little to your version of the plugin, including:
1) the pull-request changes to overcome the ImageJ API bug (I checked the update-site release of your version and it was affected)
2) the addition of a progress bar for each export, which it almost exclusively cosmetic
3) and an 'if-else' check at data import to allow workbook cells to be either string or numeric type

The plugin page on imagej.net now also indicates your involvement as well as linking to your GitHub pages:
https://imagej.net/User:ResultsToExcel

Also, I'll wait a bit before actually releasing the plugin.

bkromhout commented:
All of that sounds just fine to me. Like I said, I unfortunately won't have much time to spare to keep working on this in the future, and I certainly don't mind you taking over again :)

mayasrikanth commented:
Hello, 
Thanks for developing this plugin! I tried using Read_and_Write_Excel_Modified to move data in a results table created by Fiji's Cell Counter plugin, but the data didn't transfer. The odd thing is, it worked with the result table produced by the "particle analyzer" tool.  I was wondering whether you had any insight? Thank you!

antinos committed:
v1.1.1 and v1.1.2 changes

Bug fix and new features

antinos committed:
v1.1.5

Several bug fixes, mostly related to way the plugin handles empty cells.

antinos committed:
v1.1.6

Fixed a rare 'stack_results' bug that incorrectly exported data from a results table with only 1 row when the excel file also contained a single row of data.

antinos committed:
Install instr. changed for most up-to-date version

Minor edit to direct users to most up-to-date version of the plugin. Strikethroughs included for transparency (can be properly changed later)

antinos committed:
Install instr. changed for most up-to-date version

Installation instructions changed to direct users to the most p-to-date version of the plugin. Strikethroughs included for transparency (can be properly changed later perhaps).
*another minor edit for list display adjustments

antinos committed:
v1.1.7

Feature added. 'cell_ref' argument now allows data to be imported to a specified cell. User request from Stein Rørvik (steinr on forum.image.sc). NOTE: the plugin does not check to see if data is already present at the specified location, so overwriting is possible. Column headers are also not exported with the data.
